contracts/InterchainTokenService.sol

```diff
@@ -882,6 +898,31 @@ contract InterchainTokenService is
         }
     }
 
+    function _executeWithToken(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload,
+        string calldata tokenSymbol,
+        uint256 amount
+    ) internal {
+        bytes32 payloadHash = keccak256(payload);
+
+        address expressExecutor = _getExpressExecutor(commandId, sourceChain, sourceAddress, payloadHash);
+
+        if (!gateway.validateContractCallAndMint(commandId, sourceChain, sourceAddress, payloadHash, tokenSymbol, amount))
```

Comment:
check before getter

Reply:
slither does not like doing it that way, i could ignore it. (it's because we emit an event at the getter, and it wants the event to be emitted before everything else.

Reply:
can ignore the slither warning